device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt

```diff
@@ -371,12 +372,31 @@ class CalendarDelegate : PluginRegistry.RequestPermissionsResultListener {
         val values = ContentValues()
         val duration: String? = null
         values.put(Events.ALL_DAY, event.allDay)
-        values.put(Events.DTSTART, event.start!!)
+
         if (event.allDay) {
-            values.put(Events.DTEND, event.start!!)
+            // All day events must have UTC timezone
+            val utcTimeZone =  TimeZone.getTimeZone("UTC")
+
+            val calendar = java.util.Calendar.getInstance(utcTimeZone)
+            calendar.timeInMillis = event.start!!
+            calendar.set(java.util.Calendar.HOUR, 0)
+            calendar.set(java.util.Calendar.MINUTE, 0)
+            calendar.set(java.util.Calendar.SECOND, 0)
+            calendar.set(java.util.Calendar.MILLISECOND, 0)
+
+            val allDayTime = calendar.timeInMillis + DateUtils.DAY_IN_MILLIS
```

Comment:
Not sure this is required.

Reply:
The logic was a little wrong (I had to assign the timezone for `calendar` object after assigning the time), updated and tested successfully